### PR TITLE
Update FileStatus field names in HostModel tests

### PR DESCRIPTION
**Pull request by am11**
In https://github.com/dotnet/runtime/commit/6a81cfa4e84ab55c26abb25d82ed7355c134a602, we changed `FileSystem._fileStatus._fileStatus.Mode` to `FileSystem._fileStatus._fileCache.Mode` and https://github.com/dotnet/runtime/commit/159a01acabc79bfeb77d8c0b28de2d5fdb7773d0 adapted the change in test code during the infra update. This PR is updating field and type names along with exception message, that reflects current state.

### CODE REVIEW

File: src/installer/tests/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.AppHost.Tests/AppHostUpdateTests.cs [hunk 1]
```diff
@@ -375,7 +375,7 @@ private string PrepareAppHostMockFile(TestDirectory testDirectory, Action<byte[]
 
         private static int GetFilePermissionValue(string path)
         {
-            var modeValue = CoreFxFileStatusProvider.GetFileMode(path);
+            var modeValue = FileStatusProvider.GetFileMode(path);
```
Comment (am11) [inline src/installer/tests/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.AppHost.Tests/AppHostUpdateTests.cs:378]:
```suggestion
            int modeValue = FileStatusProvider.GetFileMode(path);
```

**Review by vitek-karas — APPROVED** [review]
Looks good - thanks a lot!

**Comment by vitek-karas** [comment]
@am11 - thanks a lot!

**Comment by am11** [comment]
Thanks! If/when FileStatus public proposal (#24700) gets approved and implemented (with real mode value on Unix), this is the first place I'd like to use new API and get rid of the private reflection. :)

**Pull request merged**